### PR TITLE
Values yaml

### PR DESCRIPTION
**Pull request by brendarearden**
**Problem**
When a values yaml was being passed through the cli, it would set the answers but when the template was getting edited in the ui, it would update the valuesYaml

**Solution**
Set valuesYaml when value is being used instead of answers

**Issue**
https://github.com/rancher/rancher/issues/23410
https://github.com/rancher/rancher/issues/21083

### CODE REVIEW

**Comment by mrajashree** [comment]
@brendarearden it looks like your PR would also fix this bug? I was considering a similar approach for it https://github.com/rancher/rancher/issues/21083#issuecomment-554076368

File: cmd/app.go [hunk 1]
```diff
@@ -433,16 +433,25 @@ func appUpgrade(ctx *cli.Context) error {
...
+		answers, err = processAnswers(ctx, c, nil, answers, false)
+		if err != nil {
+			return err
+		}
+	}
+
+	values := app.ValuesYaml
+	if ctx.String("values") != "" {
+		values, err = parseFileToString(ctx.String("values"))
+		answers = nil
```
Comment (mrajashree) [inline cmd/app.go:446]: It looks like the helm controller accepts both, answers and values and appends the passed answers to values as per https://github.com/rancher/rancher/blob/master/pkg/controllers/user/helm/common/common.go#L216-L240
So we don't need to set answers to nil here?

File: cmd/app.go [hunk 2]
```diff
@@ -433,16 +433,25 @@ func appUpgrade(ctx *cli.Context) error {
...
+	if ctx.String("answers") != "" {
+		answers, err = processAnswers(ctx, c, nil, answers, false)
+		if err != nil {
+			return err
+		}
+	}
+
+	values := app.ValuesYaml
+	if ctx.String("values") != "" {
+		values, err = parseFileToString(ctx.String("values"))
```
Comment (mrajashree) [inline cmd/app.go:445]: is it possible to extract this logic in a separate function and call that here and in templateInstall as it seems common to both?

File: cmd/app.go (hunk 1 above)
Comment (brendarearden) [reply cmd/app.go]: based on how the ui works - we should return answers or values, not both

**Comment by brendarearden** [comment]
@mrajashree I should fix that bug you linked

File: cmd/app.go [hunk 3]
```diff
@@ -1093,17 +1093,9 @@ func processAnswers(
 		answers = make(map[string]string)
 	}
 
-	if ctx.String("values") != "" {
-		if err := parseValuesFile(ctx.String("values"), answers); err != nil {
-			return answers, err
-		}
-	}
-
-	if ctx.String("answers") != "" {
```
Comment (cbron) [inline cmd/app.go:1102]: Why remove the check ?

Comment (cbron) [reply cmd/app.go:1102]: nvm

File: cmd/app.go [hunk 4]
```diff
@@ -1308,3 +1299,24 @@ func checkShowSubquestionIf(q managementClient.Question, answers map[string]stri
 	}
 	return false
 }
+
+// return answers and empty values or values and empty answers to match UI flow
```
Comment (cbron) [inline cmd/app.go:1303]: Couldn't both be returned here ? Or would both not be passed in

Comment (brendarearden) [reply cmd/app.go:1303]: I will make the comment more clear, but essentially if you pass in answers OR values, the other one should be empty, if you pass in BOTH in the cli command - both should return values

Comment (cbron) [reply cmd/app.go:1303]: Thats more clear, although code looks like if you passed both it would return both, thats why comment confused me.

**Pull request merged**